rclcpp_components/src/component_container_mt.cpp

```diff
@@ -21,8 +21,27 @@
 int main(int argc, char * argv[])
 {
   /// Component container with a multi-threaded executor.
-  rclcpp::init(argc, argv);
-  auto exec = std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
+  auto vargv = rclcpp::init_and_remove_ros_arguments(argc, argv);
+  size_t number_of_threads{0};
+  rclcpp::Logger logger{rclcpp::get_logger("component_container_mt")};
+
+  if (!vargv.empty()) {
+    for (auto itr = vargv.begin(); itr != vargv.end() - 1; ++itr) {
+      if (*itr == "--thread-num" || *itr == "-t") {
+        try {
+          number_of_threads = static_cast<size_t>(std::stoi(*(itr + 1)));
```

Comment:
what if user accidentally have `--thread-num -10`, it will try to create so many threads in `MultiThreadedExecutor`. i think that case should be avoided.

Reply:
Fixed at https://github.com/ros2/rclcpp/pull/1693/commits/9e89651ba95fc4076309c104786edf69cf65fd56